src/datastructures/Entity.java

```diff
@@ -0,0 +1,77 @@
+package datastructures;
+
+import java.util.Objects;
+import java.util.ArrayList;
+
+public class Entity {
+	String name;
+	ArrayList<String> attributes;
+
+	//*********************************************************//
+	// Constructor //
+	//*********************************************************//
+	
+	public Entity(String newname) {
+		name = newname;
+		attributes = new ArrayList<String>();
+	}
+	
+	//*********************************************************//
+	// Accessors & Mutator //
+	//*********************************************************//
+	
+	public String getName()
+	{
+		return name;
+	}
+	
+	public void setName(String newname)
+	{
+		name = newname;
+	}
+	
+	public ArrayList<String> getAttributes()
+	{
+		return attributes;
+	}
+	
+	//*********************************************************//
+	// Equals Function //
+	//*********************************************************//
+	
+	public boolean equals(Object obj) {
+		// Instance check
+		if (obj instanceof Entity)
+		{
+			// Equality check
+			Entity other = (Entity) obj;
+			return Objects.equals(this.name, other.name);
+		}
+		return false;
+	}
+
+	//*********************************************************//
+	// Attribute Functions //
+	//*********************************************************//
+	
+	public boolean createAttribute(String attribute) {
+		if (attributes.contains(attribute))
+		{
+			return true;
+		}
+		return attributes.add(attribute);
+	}	
```

Comment:
Should this return false under the if statement? Because an attribute with that name already exists, we aren't adding one so we would return false, right?

Reply:
I think in the haste of writing stuff I forgot to make that true into a false. So attrubutes.add will return a boolean, so that succeeds it will be true, false otherwise. So changing the true into a false inside of the if statement should do it.